### DISCUSSION TITLE
User roles and jobrunr

### DISCUSSION THREAD

**Discussion by valentinhttr** [jobrunr/jobrunr#774] [Q&A]
Hi,
I'd like to enqueue a job for generating my salary slips. To do this, I'll use Stream. However, in my JobRequest, I `@Autowired` my `SalarySlipService` that checks for the logged-in user roles in some methods.

And the problem is that Jobrunr has no idea about the `SecurityContextHolder` in the current job. Is it possible to create/specify it in each job for the current user?

I also found [DelegatingSecurityContextRunnable](https://docs.spring.io/spring-security/reference/servlet/integrations/concurrency.html#_delegatingsecuritycontextrunnable), but I don't see how it could be used with Jobrunr.

A small example would be welcome! 

Thanks a lot

**Comment by rdehuyss** [comment]
Hi @valh1996 , my gut feeling says the DelegatingSecurityContextRunnable won't work. It's been a long time since I've worked with Spring Security.

If I would add support for this in JobRunr Pro, do you think you could convince your manager/customer to take a Pro license?
